rows.go

```diff
@@ -638,7 +657,13 @@ func (rs *namedStructRowScanner) appendScanTargets(dstElemValue reflect.Value, s
 				colName = sf.Name
 			}
 			fpos := fieldPosByName(fldDescs, colName)
-			if fpos == -1 || fpos >= len(scanTargets) {
+			if fpos == -1 {
+				if rs.lax {
+					continue
+				}
+				return nil, fmt.Errorf("cannot find field %s in returned row", colName)
+			}
+			if fpos >= len(scanTargets) && !rs.lax {
```

Comment:
ah perhaps I am missing case where `fpos >= len(scanTargets) && rs.lax`, but I don't see how `fpos >= len(scanTargets)` is possible in the first place.